vald/evm/rpc/ethereum.go

```diff
@@ -0,0 +1,72 @@
+package rpc
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type ethereumClient struct {
+	*ethclient.Client
```

Comment:
do we need the client? seems like the only method we use is `BlockNumber`, which can be replicated by call context

Reply:
There's `TransactionReceipt(...)` as well.

Reply:
In the master branch, they've also implemented getting block with the `finalized` tag so I say we keep it around.